docs/blockentities/container.md

```diff
@@ -0,0 +1,250 @@
+
+`BaseContainerBlockEntity` bundles all calls we would normally make to our `NonNullList<ItemStack>` through two methods `#getItems` and `#setItems`, drastically reducing the amount of boilerplate we need to write. An example implementation of a `BaseContainerBlockEntity` could look like this:
+
+```java
+public class MyBlockEntity extends BaseContainerBlockEntity {
+    // The container size. This can of course be any value you want.
+    public static final int SIZE = 9;
+    // Our item stack list. This is not final due to #setItems existing.
+    private NonNullList<ItemStack> items = NonNullList.withSize(SIZE, ItemStack.EMPTY);
+
+    // The constructor, like before.
+    public MyBlockEntity(BlockPos pos, BlockState blockState) {
+        super(MY_BLOCK_ENTITY.get(), pos, blockState);
+    }
+
+    // The container size, like before.
+    @Override
+    public int getContainerSize() {
+        return SIZE;
+    }
+
+    // The getter for our item stack list.
+    @Override
+    protected NonNullList<ItemStack> getItems() {
+        return items;
+    }
+
+    // The setter for our item stack list.
+    @Override
+    protected void setItems(NonNullList<ItemStack> items) {
+        this.items = items;
+    }
+
+    // The display name of the menu. Don't forget to add a translation!
+    @Override
+    protected Component getDefaultName() {
+        return Component.translatable("container.examplemod.myblockentity");
+    }
+
+    // The menu to create from this container. See below for what to return here.
+    @Override
+    protected AbstractContainerMenu createMenu(int containerId, Inventory inventory) {
+        return null;
+    }
+}
+```
+
+Keep in mind that this class is a `BlockEntity` and a `Container` at the same time. This means that you can use the class as a supertype for your block entity to get a functioning block entity with a pre-implemented container.
+
+## Using Containers
+
+Now that we have created containers, let's use them!
+
+Since there is a considerable overlap between `Container`s and `BlockEntity`s, containers are best retrieved by casting the block entity to `Container` if possible:
+
+```java
+if (blockEntity instanceof Container container) {
+    // do something with the container
+}
+```
+
+The container can then use the methods we mentioned before, for example:
+
+```java
+// Get the first item in the container.
+ItemStack stack = container.getItem(0);
+
+// Set the first item in the container to dirt.
+container.setItem(0, new ItemStack(Items.DIRT));
+
+// Removes a quantity of (up to) 16 from the third slot.
+container.removeItem(2, 16);
+```
+
+:::warning
+A container may throw an exception if trying to access a slot that is beyond its container size.
```

Comment:
A bit misleading. For most implementations, an exception will be thrown. This only doesn't happen for `SimpleContainer` as it will return an empty stack.